service/controller/v13/adapter/recordsets.go

```diff
@@ -1,32 +1,69 @@
 package adapter
 
+import (
+	"github.com/giantswarm/aws-operator/service/controller/v13/cloudformation"
+	"github.com/giantswarm/aws-operator/service/controller/v13/key"
+	"github.com/giantswarm/microerror"
+)
+
 // The template related to this adapter can be found in the following import.
 //
 //     github.com/giantswarm/aws-operator/service/controller/v13/templates/cloudformation/guest/recordsets.go
 //
 
 type recordSetsAdapter struct {
-	APIELBHostedZones         string
-	APIELBDomain              string
-	EtcdELBHostedZones        string
-	EtcdELBDomain             string
-	IngressELBDNS             string
-	IngressELBHostedZones     string
-	IngressELBAliasHostedZone string
-	IngressELBDomain          string
-	IngressWildcardELBDomain  string
-	Route53Enabled            bool
+	BaseDomain                 string
+	GuestHostedZoneNameServers string
+	Route53Enabled             bool
 }
 
 func (r *recordSetsAdapter) getRecordSets(cfg Config) error {
-	r.APIELBHostedZones = cfg.CustomObject.Spec.AWS.API.HostedZones
-	r.APIELBDomain = cfg.CustomObject.Spec.Cluster.Kubernetes.API.Domain
-	r.EtcdELBHostedZones = cfg.CustomObject.Spec.AWS.Etcd.HostedZones
-	r.EtcdELBDomain = cfg.CustomObject.Spec.Cluster.Etcd.Domain
-	r.IngressELBHostedZones = cfg.CustomObject.Spec.AWS.Ingress.HostedZones
-	r.IngressELBDomain = cfg.CustomObject.Spec.Cluster.Kubernetes.IngressController.Domain
-	r.IngressWildcardELBDomain = cfg.CustomObject.Spec.Cluster.Kubernetes.IngressController.WildcardDomain
+	r.BaseDomain = key.BaseDomain(cfg.CustomObject)
 	r.Route53Enabled = cfg.Route53Enabled
 
 	return nil
 }
+
+func (r *recordSetsAdapter) getHostPostRecordSets(cfg Config) error {
+	var err error
+
+	err = r.getRecordSets(cfg)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	// When Route53 isn't enabled HostedZone isn't created and there are no
+	// outputs.
+	if !r.Route53Enabled {
+		return nil
+	}
+
+	var cf *cloudformation.CloudFormation
+	{
+		c := cloudformation.Config{
+			Client: cfg.Clients.CloudFormation,
+		}
+
+		cf, err = cloudformation.New(c)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
```

Comment:
This is not cool because it works around the resource implementation. It would be better to inject the outputs we need to the adapter magic. We already fetched them in `GetCurrentState`. There is probably also the point where cancelation is already done so I am not sure which purpose the condition above serves. I also do not like that we create a client here over and over again within the depths of the code. There should be an earlier entrypoint. AFAIK there is the controller context magic we use for BYOC. That should be aligned with that somehow.